server/src/http/feature_refresher.rs

```diff
@@ -80,16 +81,20 @@ fn update_projects_from_feature_update(
         vec![]
     } else {
         let projects_to_update = &token.projects;
-        let mut to_keep: Vec<ClientFeature> = original
-            .iter()
-            .filter(|toggle| {
-                let p = toggle.project.clone().unwrap_or_else(|| "default".into());
-                !projects_to_update.contains(&p)
-            })
-            .cloned()
-            .collect();
-        to_keep.extend(updated.iter().cloned());
-        to_keep
+        if projects_to_update.contains(&"*".into()) {
+            updated.into()
+        } else {
+            let mut to_keep: Vec<ClientFeature> = original
+                .iter()
+                .filter(|toggle| {
+                    let p = toggle.project.clone().unwrap_or_else(|| "default".into());
+                    !projects_to_update.contains(&p)
+                })
+                .cloned()
+                .collect();
+            to_keep.extend(updated.iter().cloned());
+            to_keep.iter().unique_by(|cf| &cf.name).cloned().collect()
```

Comment:
Don't think this is needed, we should now be deleting every project the incoming token has access to, which means extending with the update should be just fine without eliminating duplicates.

Reply:
I'll remove it again.